src/minefield.rs

```diff
@@ -175,10 +248,23 @@ impl Board {
     /// The `skip` argument contains board indices
     /// that shall not have a mine placed in.
     fn place_mines(&mut self, skip: &[usize]) {
-        todo!()
+        // i would put (usize, usize) here, since its just one point user clicks on + eventual flood
+        // and then have this method be called in handle_uncover at the beginning
+        self.placed = true;
+        let mut rng = thread_rng();
+        let idx_range = Range { start: 0, end: self.params.height * self.params.height };
+
+        for i in 0..self.params.mines {
+            while let Some(mine_idx) = idx_range.clone().choose(&mut rng) {
+                if skip.contains(&mine_idx) {
+                    continue;
+                }
+                self.tiles[mine_idx].object = Object::Mine;
+            }
+        }
```

Comment:
I was thinking about something more like:

```rust
(0..self.tiles.len()).filter(|i| !skip.contains(i)).choose_multiple(&mut rng, self.params.mines)
```

Reply:
(Also, this particular implementation might choose the same index multiple times and then we end up with fewer mines than required)

Reply:
Done. Used your example implementation with a twist.